packages/nimble-components/src/utilities/unit-format/decimal-unit-format.ts

```diff
@@ -0,0 +1,76 @@
+import { UnitFormat } from './base/unit-format';
+import type { ScaledUnitFormat } from './unit-scale/base/scaled-unit-format';
+import type { UnitScale } from './unit-scale/base/unit-scale';
+import { passthroughUnitScale } from './unit-scale/passthrough-unit-scale';
+
+/**
+ * Format for decimal numbers with units.
+ */
+export class DecimalUnitFormat extends UnitFormat {
+    /** Resolved UnitScale */
```

Comment:
(this comment resolves multiple discussions)
Goal from the change was to align with `Intl.NumberFormat` api as stated in the README so that leads to two changes:
- constructor takes two params, locale and options object, so that was done for both UnitFormats even if today it only has one option: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#parameters (added some types to emphasize that)
- the constructor params are resolved to different values and you can observe those changes in the public api (added resolvedOptions to match `Intl.NumberFormat` api instead of individual properties)